File: android/app/src/main/java/edu/gatech/cs2340/rajsjarfiles/spacetrader/entity/player/action/BribeAction.java
package edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.player.action;

import edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.player.Player;
import edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.player.Ship;

/**
 * Represents a player's bribe action.
 */
public class BribeAction implements PlayerEncounterAction {

    @Override
    public String doAction(Player player, Ship otherShip, EncounterState es) {
        String retString = "";

        retString += "You bribe the police with 50% of your money.\n";
        retString += "You had "
                + player.getCredits() + " credits.\n";
        int newCredits = player.setCredits(0.5f);
        retString += "You now have "
                + newCredits + " credits.\n";

        es.conclude();

        return retString;
    }
}

File: ios/SpaceTrader/SpaceTrader/src/entity/player/action/BribeAction.swift
//
//  .swift
//  SpaceTrader
//
//  Created by Thomas Suarez on 4/20/19.
//  Copyright © 2019 RajsJarFiles. All rights reserved.
//

import Foundation

class BribeAction : PlayerEncounterAction {
    
    func doAction(player: Player, otherShip: Ship, es: EncounterState) -> String {
        var retString = ""
        
        retString += "You bribe the police with 50% of your money.\n"
        retString += "You had \(player.wallet.credits) credits.\n"
        player.wallet.credits /= 2
        retString += "You now have \(player.wallet.credits) credits.\n"
        
        es.conclude()
        
        return retString
    }
    
}
